UI/CUIControls.h

```diff
@@ -594,4 +595,48 @@ class RotatingGraphic : public GG::StaticGraphic {
     float   m_phase_offset;
 };
 
+class CUIToggleButton;
+
+/** \brief A FreeOrion styled ToggleButton. */
+class CUIToggleButtonRepresenter : public GG::StateButtonRepresenter {
+public:
+    virtual void   Render(const CUIToggleButton& button) const;
+
+    virtual void   OnChecked(bool checked) const;
+
+    virtual GG::Pt MinUsableSize(const CUIToggleButton& button) const;
+};
+
+/** \brief Toggled StateButton
+ * 
+ * This StateButton renders a SubTexture depending on the checked state.
+ * With the default single texture it renders the same icon recolored towards grey.
+ * When a second SebTexture is assigned, it will render the appropriate SubTexture.
+ */
+class CUIToggleButton : public GG::StateButton {
```

Comment:
Yeah, but how is this an argument against my own request? `Should` is not `must` also the comment is written under the prerequisite that the ButtonRepresenter instance is shared, which isn't the case for the image variant. If there is a valid reason to violate the rule (here: there doesn't exist a storage for storing instance specific rendering data yet), then do it.